lib/StackAnalysis/StackAnalysis.cpp

```diff
@@ -383,6 +433,811 @@ void commitToModel(GeneratedCodeBasicInfo &GCBI,
+  IRBuilder<> Builder(&OutFunc->getEntryBlock().front());
+  auto *IntPtrTy = GCBI->spReg()->getType();
+  auto *IntTy = GCBI->spReg()->getType()->getElementType();
+
+  auto *SP = Builder.CreateLoad(GCBI->spReg());
+  auto *SPPtr = Builder.CreateIntToPtr(SP, IntPtrTy);
+  auto *GEPI = Builder.CreateGEP(IntTy, SPPtr, ConstantInt::get(IntTy, 0));
+
+  switch (GCBI->arch()) {
+  case Triple::arm: {
+    constexpr auto *LinkRegister = "r14";
+    RA = Builder.CreateLoad(M.getGlobalVariable(LinkRegister, true));
+    break;
+  }
+
+  case Triple::aarch64: {
+    constexpr auto *LinkRegister = "lr";
+    RA = Builder.CreateLoad(M.getGlobalVariable(LinkRegister, true));
+    break;
+  };
+
+  case Triple::mips:
+  case Triple::mipsel: {
+    constexpr auto *ReturnRegister = "ra";
+    RA = Builder.CreateLoad(M.getGlobalVariable(ReturnRegister, true));
+    break;
+  }
+
+  case Triple::systemz: {
+    constexpr auto *ReturnRegister = "r14";
+    RA = Builder.CreateLoad(M.getGlobalVariable(ReturnRegister, true));
+    break;
+  }
+
+  case Triple::x86:
+  case Triple::x86_64: {
+    RA = Builder.CreateLoad(SPPtr);
+    break;
+  }
+
+  default:
+    revng_abort("Unsupported architecture");
+  }
+
+  std::array<Value *, 4> DissectedPC = PCH->dissectJumpablePC(Builder,
+                                                              RA,
+                                                              GCBI->arch());
+
+  auto *PCI = MetaAddress::composeIntegerPC(Builder,
+                                            DissectedPC[0],
+                                            DissectedPC[1],
+                                            DissectedPC[2],
+                                            DissectedPC[3]);
+
+  SmallVector<Value *, 16> CSVI;
+  Type *IsRetTy = Type::getInt128Ty(Context);
+  SmallVector<Type *, 16> ArgTypes = { IsRetTy, IntTy, MetaAddressTy };
+  for (auto *CSR : ABIRegisters) {
+    auto *V = Builder.CreateLoad(CSR);
+    CSVI.emplace_back(V);
+    ArgTypes.emplace_back(IntTy);
+  }
+
+  IndirectBranchInfoMarker = Function::Create(llvm::FunctionType::get(IntTy,
+                                                                      ArgTypes,
+                                                                      false),
+                                              GlobalValue::ExternalLinkage,
+                                              "indirect_branch_info",
+                                              M);
+  IndirectBranchInfoMarker->addFnAttr(Attribute::NoUnwind);
+  IndirectBranchInfoMarker->addFnAttr(Attribute::NoReturn);
+
+  SmallVector<BasicBlock *, 1> IndirectBranches = { AnyPCCloned };
+  SmallVector<Instruction *, 4> BranchesForIBI;
+  for (auto *IB : IndirectBranches) {
+    if (IB) {
+      for (auto *Pred : predecessors(IB)) {
+        auto *Term = Pred->getTerminator();
+        revng_assert(isa<BranchInst>(Term));
+        BranchesForIBI.emplace_back(Term);
+      }
+    }
+  }
```

Comment:
Why are we iterating over a vector that has a single element?

Reply:
Because originally the vector contained `UnexpectedPCCloned` as well, removing the vector.